core/packages/contracts/src/Gateway.sol

```diff
@@ -384,10 +384,7 @@ contract Gateway is IGateway {
 
     // Register a token on AssetHub
     function registerToken(address token) external payable {
-        CoreStorage.Layout storage $ = CoreStorage.layout();
-        address assetHubAgent = $.agents[ASSET_HUB_AGENT_ID];
-
-        (bytes memory payload, uint256 extraFee) = Assets.registerToken(assetHubAgent, token, CREATE_TOKEN_CALL_ID);
+        (bytes memory payload, uint256 extraFee) = Assets.registerToken(address(this), token, CREATE_TOKEN_CALL_ID);
```

Comment:
```suggestion
        (bytes memory payload, uint256 extraFee) = Assets.registerToken(token, CREATE_TOKEN_CALL_ID);
```

Reply:
addressed in https://github.com/Snowfork/snowbridge/pull/904/commits/8289362594510c9672b87bd44bfbf09dcbe3f596